### PR TITLE
backend: Update Class Model to include Student field

### PR DESCRIPTION
**Pull request by wais-hundekar**
## Notion ticket link

[Update Class Model to Include Student Field](https://www.notion.so/uwblueprintexecs/47bf4727a2444bccad4a91952898b8f3?v=9658436bf9be409e90581631a7155f16&p=3d65b934711440dab26e3b9657de7e15&pm=s)

## Implementation description
* Adding Student Field and Student Schema to Class Model

## Steps to test
1. Ensure all tests pass and no functionality is broken

## What should reviewers focus on?
* Ensure all required parts of the code are updated, there is no ClassService so I assume this should suffice.

## Checklist
- [ ] My PR name is descriptive and in imperative tense
- [ ] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [ ] I have run the appropriate linter(s)
- [ ] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: backend/models/class.model.ts [hunk 1]
```diff
@@ -17,6 +17,14 @@ export interface Class extends Document {
   teacher: string;
   /** the ids of the test sessions assigned to the class */
   testSessions: string[];
+  /** the students of the class */
+  students: Student[];
+}
+
+export interface Student {
```
Comment (carissa-tang) [inline backend/models/class.model.ts]: could you actually make a Student schema too? and Student should have an `id` field

Comment (carissa-tang) [reply backend/models/class.model.ts]: you can reference ResultSchema in testSession.model.ts as an example, except your Student interface needs to have the id too

Comment (wais-hundekar) [reply backend/models/class.model.ts]: Changed!

File: backend/models/class.model.ts [hunk 2]
```diff
@@ -17,8 +17,36 @@ export interface Class extends Document {
...
+  firstName: string;
+  lastName: string;
+  studentNumber?: string;
+}
+
+const StudentSchema: Schema = new Schema({
+  id: {
+    type: Number,
+    required: true,
+  },
```
Comment (carissa-tang) [inline backend/models/class.model.ts]:
```suggestion
```
id by default is added to all schemas!

**Review by carissa-tang — APPROVED** [review]
you can merge after removing the id!

**Pull request merged**